ChatSX/screens/HomeFeed/Tabs/DMFeed/index.tsx

```diff
@@ -99,6 +106,157 @@ const DMFeed = ({navigation}: Props) => {
     token();
   }, []);
 
+  // Sync Chatrrom API
+  async function syncChatroomAPI(
+    page: number,
+    minTimeStamp: number,
+    maxTimeStamp: number,
+  ) {
+    const res = await myClient?.syncChatroom(
+      SyncChatroomRequest.builder()
+        .setPage(page)
+        .setPageSize(50)
+        .setChatroomTypes([10])
+        .setMaxTimestamp(maxTimeStamp)
+        .setMinTimestamp(minTimeStamp)
+        .build(),
+    );
+    return res;
+  }
+
+  // Pagination call for sync chatroom
+  const paginatedSyncAPI = async (page: number, communityId: string) => {
+    const timeStampStored = await myClient?.getTimeStamp();
+    const temp = JSON.stringify(timeStampStored);
+    let parsedTimeStamp = JSON.parse(temp);
+    let maxTimeStampNow = Math.floor(Date.now() / 1000);
+
+    // Taking minTimeStamp as 0 for the first time else last maxTimeStamp will become current minTimeStamp
+    let minTimeStampNow =
+      parsedTimeStamp[0].minTimeStamp === 0
+        ? 0
+        : parsedTimeStamp[0].maxTimeStamp;
+
+    const val = await syncChatroomAPI(page, minTimeStampNow, maxTimeStampNow);
+
+    const DB_RESPONSE = val?.data;
+
+    if (page === INITIAL_SYNC_PAGE && DB_RESPONSE?.chatroomsData.length !== 0) {
+      await myClient?.saveCommunity(DB_RESPONSE?.communityMeta[communityId]);
+    }
+
+    if (DB_RESPONSE?.chatroomsData.length !== 0) {
+      const totalChatrooms = DB_RESPONSE?.chatroomsData;
+      for (let i = 0; i < totalChatrooms.length; i++) {
+        const chatroom = totalChatrooms[i];
+        const userData =
+          user?.id !== chatroom?.chatroomWithUserId
+            ? DB_RESPONSE?.userMeta[chatroom?.chatroomWithUserId]?.name
+            : DB_RESPONSE?.userMeta[chatroom?.userId]?.name;
+
+        DB_RESPONSE.chatroomsData[i].chatroomWithUserName = userData;
+      }
```

Comment:
move this part to ROConvertor

Reply:
can't move this part to `ROConverter` as here I am using `user` which comes from Redux